File: Programmers/타겟_넘버.java
class Solution {
    static int answer = 0;

    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return answer;
    }
    
    public static void dfs(int[] numbers, int target, int sum, int idx) {
        if (idx >= numbers.length) {
            if (sum == target) answer++;
            return;
        }

        dfs(numbers, target, sum + numbers[idx], idx + 1);
        dfs(numbers, target, sum - numbers[idx], idx + 1);
    }
}

File: Programmers_python/타겟 넘버.py
answer = 0

def dfs(numbers, idx, total, target):
    global answer
    
    if idx == len(numbers):
        if total == target:
            answer += 1
        return 
        
    dfs(numbers, idx+1, total + numbers[idx], target)
    dfs(numbers, idx+1, total - numbers[idx], target)
    
def solution(numbers, target):
    dfs(numbers, 0, 0, target)
    
    return answer
